src/rpp/stack_trace.cpp

```diff
@@ -1129,6 +1144,36 @@ namespace rpp
         return (int)count;
     }
 
+    RPPAPI std::vector<ThreadCallstack> get_all_callstacks()
+    {
+        std::vector<ThreadCallstack> threads {};
+
+        HANDLE hSnap = CreateToolhelp32Snapshot(TH32CS_SNAPTHREAD, 0);
+        if (hSnap == INVALID_HANDLE_VALUE) return threads;
+
+        THREADENTRY32 te;
+        te.dwSize = sizeof(te);
+
+        const DWORD currentProcessId = GetCurrentProcessId();
+
+        if (Thread32First(hSnap, &te))
+        {
+            do {
+                if (te.th32OwnerProcessID != currentProcessId) continue;
+
+                std::vector<uint64_t> trace = get_callstack(CALLSTACK_MAX_DEPTH, /*entriesToSkip*/1, static_cast<uint64_t>(te.th32ThreadID));
```

Comment:
`/*entriesToSkip*/1`

This is a problem in multi-thread capture, because only the calling thread knows how many entries to skip, all other threads should not skip any entries.

So:
```
const DWORD currentThreadId = GetCurrentThreadId();
...
int entriesToSkip = te.th32ThreadID == currentThreadId ? 1 : 0;
```
It would be best if `int entriesToSkip` was also a parameter and defined main calling thread's entriesToSkip count, but for all other threads it would be 0.